### PR TITLE
Issue #2976899 by idebr: [Domain source] Follow Drupal coding standards

### PR DESCRIPTION
**Pull request by ideb**

### CODE REVIEW

**Comment by agentrickard** [comment]
These test fails are unrelated to this work.

File: domain_source/domain_source.module [hunk 1]
```diff
@@ -166,7 +169,7 @@ function domain_source_form_validate($element, \Drupal\Core\Form\FormStateInterf
 /**
  * Redirect form submissions to other domains.
  */
-function domain_source_form_submit(&$form, \Drupal\Core\Form\FormStateInterface $form_state) {
+function domain_source_form_submit(&$form, FormStateInterface $form_state) {
```
Comment (agentrickard) [inline domain_source/domain_source.module:172]: Do we have to type hint &$form?

File: domain_source/domain_source.module [hunk 2]
```diff
@@ -138,7 +141,7 @@ function domain_source_form_alter(&$form, &$form_state, $form_id) {
 /**
  * Validate form submissions.
  */
-function domain_source_form_validate($element, \Drupal\Core\Form\FormStateInterface $form_state) {
+function domain_source_form_validate($element, FormStateInterface $form_state) {
```
Comment (agentrickard) [inline domain_source/domain_source.module:144]: Do we have to type hint $element?

**Review by agentrickard — COMMENTED** [review]
Two simple questions about type hints

File: domain_source/domain_source.module (hunk 1 above)
Comment (ideb) [reply domain_source/domain_source.module:172]: FormSubmitterInterface::doSubmitForm does not typehint the &$form array, but it is assumed to be an array so I added a type cast to make this explicit

File: domain_source/domain_source.module (hunk 2 above)
Comment (ideb) [reply domain_source/domain_source.module:144]: FormValidatorInterface::executeValidateHandlers does not typehint the &$form array, but it is assumed to be an array so I added a type cast to make this explicit

**Review by agentrickard — CHANGES_REQUESTED** [review]
I would remove these changes. I looked at core, and they don't type hint these vars.

```
/**
 * Implements hook_form_BASE_FORM_ID_alter() for \Drupal\node\NodeForm.
 */
function path_form_node_form_alter(&$form, FormStateInterface $form_state) {
```

**Comment by agentrickard** [comment]
Thanks! I'll wait for the tests here and then clean up anything that breaks.

**Pull request merged**